src/vcl/FaissDescriptorSet.cc

```diff
@@ -339,3 +343,65 @@ long FaissIVFFlatDescriptorSet::add(float *descriptors, unsigned n,
 
   return id_first;
 }
+
+
+// FaissHNSWFlat
+//Note:
+//setting value of hnsw m= 48
+  // M is number of connections each vertex will have 
+  //i.e. number of nearest neighbors that each vertex will connect to.  
+  // Total memory usage is (dim * 4 + M * 2 * 4) bytes per vector.
+
+  // Rule of thumb to support 1M entries
+  // according to https://github.com/facebookresearch/faiss/wiki/Indexing-1M-vectors
+  // HNSW_M= 48
+  // efConstruction = 2 x m
+  // efSeach = 32
+  // Default values from hnsw.h
+      //int efConstruction = 40;
+      //int efSearch = 16;
+  //efsearch will be defined in the search params when IndexHNSW::search() is called (can dynamically change during runtime)
+  //efconstruction is a compile time paramerter will be defined in IndexHNSW.hnsw struct 
+
+FaissHNSWFlatDescriptorSet::FaissHNSWFlatDescriptorSet(
+    const std::string &set_path)
+    : FaissDescriptorSet(set_path) {
+  try {
+    _index = faiss::read_index(_faiss_file.c_str());
+
+  } catch (faiss::FaissException &e) {
+    throw VCLException(OpenFailed, "Problem reading: " + _faiss_file);
+  }
+
+  // Faiss will sometimes throw, or sometimes set _index = NULL,
+  // we check both just in case.
+  if (!_index) {
+    throw VCLException(OpenFailed, "Problem reading: " + _faiss_file);
+  }
+
+  _dimensions = _index->d;
+  _n_total = _index->ntotal;    
+  
+}
+
+FaissHNSWFlatDescriptorSet::FaissHNSWFlatDescriptorSet(
+    const std::string &set_path, unsigned dim, DistanceMetric metric)
+    : FaissDescriptorSet(set_path, dim) {
+
+  int hnsw_M= 48;
+  if (metric == L2) {
+        _index = new faiss::IndexHNSWFlat(dim, hnsw_M, faiss::METRIC_L2);
+        ((faiss::IndexHNSWFlat*)_index)->hnsw.efConstruction = 96;
+  } else { 
+  //only metric L2 is supported for HNSWFLAT for FAISS v1.7.4
+  //newer version of Faiss e.g. V1.8.0 supports I.P. metric for HNSW
+    throw VCLException(UnsupportedIndex, "Metric Not implemented");
+  }
+  
+}
+
+void FaissHNSWFlatDescriptorSet::search(float *query, unsigned n_queries, unsigned k, long *descriptors, float *distances)  {
+  ((faiss::IndexHNSWFlat*)_index)->hnsw.efSearch = 128;
```

Comment:
a comment on why efSearch is set to 128 would be helpful

Reply:
added a comment to the logic behind setting the different index parameters in the latest commit